hooks/secondary_publish_tk-mari.py

```diff
@@ -223,21 +223,33 @@ def _publish_textures(self, item, output, sg_task, comment, thumbnail_path, prog
             progress_cb(50, "Exporting layer")
             layer.exportImages(output_path)
         else:
-            # flatten layers in the channel and publish the flattened layer:
+            # publish the entire channel, flattened
             progress_cb(50, "Exporting channel")
-            # remember the current channel:
-            current_channel = geo.currentChannel()
-            # duplicate the channel so we don't operate on the original:
-            duplicate_channel = geo.createDuplicateChannel(channel)
-            # flatten it into a single layer:
-            flattened_layer = duplicate_channel.flatten()
-            # export the images for it:
-            flattened_layer.exportImages(output_path)
-            # set the current channel back - not doing this will result in Mari crashing
-            # when the duplicated channel is removed!
-            geo.setCurrentChannel(current_channel)
-            # remove the duplicate channel, destroying the channel and the flattened layer:
-            geo.removeChannel(duplicate_channel, geo.DESTROY_ALL)
+            layers = channel.layerList()
+            if len(layers) == 1:
+                # only one layer so just publish it:
+                # Note - this works around an issue that was reported (#27945) where flattening a channel
+                # with only a single layer would cause Mari to crash - this bug was not reproducible by
+                # us but happened 100% for the client!
+                layer = layers[0]
+                layer.exportImages(output_path)
+            elif len(layers) > 0:
```

Comment:
Although the code will do the right thing with the current structure, wouldn't it be better to have `elif len(layers) > 1:` to highlight that this block is valid only if we have more than one layer ?